parser/prism/Translator.cc

```diff
@@ -757,58 +761,15 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
         case PM_MULTI_TARGET_NODE: { // A multi-target like the `(x2, y2)` in `p1, (x2, y2) = a`
             auto multiTargetNode = down_cast<pm_multi_target_node>(node);
 
-            auto prismLefts = absl::MakeSpan(multiTargetNode->lefts.nodes, multiTargetNode->lefts.size);
-            auto prismRights = absl::MakeSpan(multiTargetNode->rights.nodes, multiTargetNode->rights.size);
-            auto prismRestNode = multiTargetNode->rest;
-
-            NodeVec sorbetExpressions{};
-            sorbetExpressions.reserve(prismLefts.size() + prismRights.size() + (prismRestNode != nullptr ? 1 : 0));
-
-            translateMultiInto(sorbetExpressions, prismLefts);
-
-            if (prismRestNode != nullptr) {
-                // We don't just call `translate()` because that would create a `parser::Splat`, but we need
-                // `parser::SplatLhs`.
-                auto splatNode = down_cast<pm_splat_node>(prismRestNode);
```

Comment:
This was an incorrect assumption, and used to be unchecked prior to #302. This variable can now be a `PM_IMPLICIT_REST_NODE`, and is handled correctly.

https://github.com/Shopify/sorbet/blob/ee541d10ae57b2bc471771c054fbd689eb6ba643/parser/prism/Translator.cc#L799-L802